### PR TITLE
Development updates for DFG v0.6

### PR DESCRIPTION
**Pull request by Affie**

### CODE REVIEW

File: src/CliqStateMachineUtils.jl [hunk 1]
```diff
@@ -898,7 +898,8 @@ directPriorMsgIDs, directFrtlMsgIDs, directAssignmentIDs, mcmcIterationIDs
 function determineCliqVariableDownSequence(subfg::AbstractDFG, cliq::TreeClique; solvable::Int=1)
   frtl = getCliqFrontalVarIds(cliq)
 
-  adj = DFG.getAdjacencyMatrix(subfg, solvable=solvable)
+  #TODO don't use this getAdjacencyMatrixSymbols
+  adj = DFG.getAdjacencyMatrixSymbols(subfg, solvable=solvable)
```
Comment (Affie) [inline src/CliqStateMachineUtils.jl]: This has to be updated to use `getBiadjacencyMatrix`

File: src/FactorGraph01.jl [hunk 2]
```diff
@@ -552,7 +548,7 @@ end
 
 Fetch the variable marginal sample points without the KDE bandwidth parameter.  Use getVertKDE to retrieve the full KDE object.
 """
-function getVal(vA::Vector{DFGVariable}, solveKey::Symbol=:default)::Array{Float64, 2}
+function getVal(vA::Vector{<:DFGVariable}, solveKey::Symbol=:default)::Array{Float64, 2}
```
Comment (Affie) [inline src/FactorGraph01.jl]: This is because a DFGVariable now takes type parameters for softtype. like DFGFactor

File: src/FactorGraph01.jl [hunk 3]
```diff
@@ -1017,19 +1015,21 @@ function addFactor!(dfg::G,
...
+  newFactor = DFGFactor(Symbol(namestring),
+                        varOrderLabels,
+                        solverData;
+                        tags=Set(union(labels, [:FACTOR])),
+                        solvable=solvable)
 
   # TODO: Need to remove this...
   for vert in Xi
-    push!(newData.fncargvID, vert.label) # vert._internalId # YUCK :/ -- Yup, this is a problem
+    push!(solverData.fncargvID, vert.label)
```
Comment (Affie) [inline src/FactorGraph01.jl]: I tried to follow through on @GearsAD changes,. Please just confirm if this is what you wanted.

File: src/FactorGraph01.jl [hunk 4]
```diff
@@ -1017,19 +1015,21 @@ function addFactor!(dfg::G,
...
   end
 
-  success = DFG.addFactor!(dfg, Xi, newFactor)
+  success = DFG.addFactor!(dfg, newFactor)
```
Comment (Affie) [inline src/FactorGraph01.jl]: Note, I recommend we use this with variable order extracted from the factor.

File: src/DispatchPackedConversions.jl [hunk 5]
```diff
@@ -135,20 +135,20 @@ completely rebuild the factor's CCW and user data.
 function rebuildFactorMetadata!(dfg::G, factor::DFGFactor)::DFGFactor where G <: AbstractDFG
   # Set up the neighbor data
   neighbors = map(vId->getVariable(dfg, vId), getNeighbors(dfg, factor))
-  neighborUserData = map(v->solverData(v).softtype, neighbors)
+  neighborUserData = map(v->getSolverData(v).softtype, neighbors)
 
   # Rebuilding the CCW
-  setDefaultFactorNode!(dfg, factor, neighbors, factor.data.fnc.usrfnc!)
+  ccw_new = getDefaultFactorData(dfg, neighbors, factor.data.fnc.usrfnc!)
+  setSolverData!(factor, ccw_new)
```
Comment (Affie) [inline src/DispatchPackedConversions.jl]: Also confirm this.

File: src/CliqStateMachineUtils.jl (hunk 1 above)
Comment (dehann) [reply src/CliqStateMachineUtils.jl]: understood

File: src/DispatchPackedConversions.jl (hunk 5 above)
Comment (dehann) [reply src/DispatchPackedConversions.jl]: i recently found this issue, might be somewhat related to your concern here: #568

File: src/FactorGraph01.jl (hunk 2 above)
Comment (dehann) [reply src/FactorGraph01.jl]: sounds good!

File: src/FactorGraph01.jl (hunk 3 above)
Comment (dehann) [reply src/FactorGraph01.jl]: defer this to @GearsAD please

File: src/FactorGraph01.jl (hunk 4 above)
Comment (dehann) [reply src/FactorGraph01.jl]: Perhaps I could ask for a bit more detail on this example please?  Willing to give it a try, as long as we provide both APIs until we decide which is the long term strategy.

Comment (Affie) [reply src/FactorGraph01.jl]: Both can stay forever. just recommend this one. So drivers should implement this one, the others are all done for AbstractDFG.

Comment (Affie) [reply src/FactorGraph01.jl]: see here: https://github.com/JuliaRobotics/DistributedFactorGraphs.jl/blob/3ec2484a2b22a00b47027a310e284ff674f08a9c/src/services/AbstractDFG.jl#L217-L236

**Pull request merged**